.github/workflows/pytest.yml

```diff
@@ -1,54 +1,89 @@
-name: pytest
+name: CI
 
 on:
   push:
     branches:
       - master
   pull_request:
 
+# Ref: https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  PYTHONDONTWRITEBYTECODE: 1
+  FORCE_COLOR: 1 # colored output by pytest etc.
+  CLICOLOR_FORCE: 1 # colored output by ruff
+
 jobs:
-  build:
-    runs-on: ubuntu-20.04
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: "3.11"
+        cache: pip
+    - run: python -m pip install --upgrade pip
+    - run: pip install -e .[dev]
+    - name: Set up pre-commit cache
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pre-commit
+        key: lint-${{ hashFiles('.pre-commit-config.yaml') }}
+    - name: Run black and ruff via pre-commit
+      run: pre-commit run --all-files --color=always
+      continue-on-error: true # TODO: remove once files are reformatted
+  
+  test:
+    runs-on: ubuntu-22.04
     strategy:
       matrix:
+        python-version: ['3.8', '3.11']
         db-backend: [mysql, postgres, sqlite3]
-        python-version: ['3.6', '3.11']
-
-    services:
-      postgres:
-        image: postgres:latest
-        env:
-          POSTGRES_DB: rdmo
-          POSTGRES_PASSWORD: postgres_password
-        ports:
-          - 5432:5432
-
+      fail-fast: false # TODO: for debugging tests only
```

Comment:
This line can be removed. It was added to show that tests fail for Python 3.6, which is now removed anyway. So CI can fail fast again.